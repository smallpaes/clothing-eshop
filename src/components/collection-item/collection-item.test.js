import React from 'react';
import { shallow } from 'enzyme';

import CollectionItem from './collection-item.component';


describe('CollectionItem component', () => {
  let wrapper;
  let mockAddItem;
  let mockItem;

  beforeEach(() => {
    mockItem = {
      name: 'Shoe', 
      price: 320, 
      imageUrl: 'www.test.com'
    };
    mockAddItem = jest.fn();
    wrapper = shallow(<CollectionItem item={mockItem} addItem={mockAddItem} />);
  });

  it('Should render CollectionItem component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('addItem is triggered when AddButton is clicked', () => {
    const addButton = wrapper.find('AddButton');
    addButton.simulate('click');
    expect(mockAddItem.mock.calls.length).toBe(1);
  });

  it('should render imageUrl as a prop on BackgroundImage', () => {
    expect(wrapper.find('BackgroundImage').prop('imageUrl')).toEqual(mockItem.imageUrl);
  });
});
