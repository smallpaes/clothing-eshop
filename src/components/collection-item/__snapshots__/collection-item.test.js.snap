// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component Should render CollectionItem component 1`] = `ShallowWrapper {}`;

exports[`Should render CollectionItem component 1`] = `ShallowWrapper {}`;
